### PR TITLE
Rustup to syntax::errors changes 

### LINKED ISSUE #525: Does not compile on latest rust nightly
```
   Compiling clippy v0.0.33 (file:///Users/coreyf/Development/rust/rust-clippy)
src/utils.rs:314:19: 316:32 error: no method named `fileline_help` found for type `&rustc::session::Session` in the current scope
src/utils.rs:314         cx.sess().fileline_help(sp, &format!("for further information visit \
src/utils.rs:315             https://github.com/Manishearth/rust-clippy/wiki#{}",
src/utils.rs:316             lint.name_lower()))
src/utils.rs:337:19: 339:38 error: no method named `fileline_help` found for type `&rustc::session::Session` in the current scope
src/utils.rs:337         cx.sess().fileline_help(span, &format!("{}\nfor further information \
src/utils.rs:338             visit https://github.com/Manishearth/rust-clippy/wiki#{}",
src/utils.rs:339             help, lint.name_lower()))
src/utils.rs:348:23: 348:53 error: no method named `fileline_note` found for type `&rustc::session::Session` in the current scope
src/utils.rs:348             cx.sess().fileline_note(note_span, note)
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/utils.rs:350:23: 350:49 error: no method named `span_note` found for type `&rustc::session::Session` in the current scope
src/utils.rs:350             cx.sess().span_note(note_span, note)
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~
src/utils.rs:352:19: 354:32 error: no method named `fileline_help` found for type `&rustc::session::Session` in the current scope
src/utils.rs:352         cx.sess().fileline_help(span, &format!("for further information visit \
src/utils.rs:353             https://github.com/Manishearth/rust-clippy/wiki#{}",
src/utils.rs:354             lint.name_lower()))
src/utils.rs:363:19: 365:32 error: no method named `fileline_help` found for type `&rustc::session::Session` in the current scope
src/utils.rs:363         cx.sess().fileline_help(sp, &format!("for further information visit \
src/utils.rs:364             https://github.com/Manishearth/rust-clippy/wiki#{}",
src/utils.rs:365             lint.name_lower()))
src/eta_reduction.rs:89:35: 91:59 error: no method named `span_suggestion` found for type `&rustc::session::Session` in the current scope
src/eta_reduction.rs:89                         cx.sess().span_suggestion(expr.span,
src/eta_reduction.rs:90                                                   "remove closure as shown:",
src/eta_reduction.rs:91                                                   snippet);
src/eta_reduction.rs:55:20: 55:63 error: lifetime bound not satisfied [E0478]
src/eta_reduction.rs:55                 if args.iter().any(|arg| is_adjusted(cx, arg)) {
                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/eta_reduction.rs:42:49: 97:2 note: lifetime parameter instantiated with the anonymous lifetime #3 defined on the block at 42:48
src/eta_reduction.rs:42 fn check_closure(cx: &LateContext, expr: &Expr) {
src/eta_reduction.rs:43     if let ExprClosure(_, ref decl, ref blk) = expr.node {
src/eta_reduction.rs:44         if !blk.stmts.is_empty() {
src/eta_reduction.rs:45             // || {foo(); bar()}; can't be reduced here
src/eta_reduction.rs:46             return;
src/eta_reduction.rs:47         }
                        ...
src/eta_reduction.rs:42:49: 97:2 note: but lifetime parameter must outlive the anonymous lifetime #2 defined on the block at 42:48
src/eta_reduction.rs:42 fn check_closure(cx: &LateContext, expr: &Expr) {
src/eta_reduction.rs:43     if let ExprClosure(_, ref decl, ref blk) = expr.node {
src/eta_reduction.rs:44         if !blk.stmts.is_empty() {
src/eta_reduction.rs:45             // || {foo(); bar()}; can't be reduced here
src/eta_reduction.rs:46             return;
src/eta_reduction.rs:47         }
                        ...
src/shadow.rs:185:23: 185:66 error: no method named `span_note` found for type `&rustc::session::Session` in the current scope
src/shadow.rs:185             cx.sess().span_note(span, "previous binding is here");
                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/returns.rs:80:27: 82:51 error: no method named `span_suggestion` found for type `&rustc::session::Session` in the current scope
src/returns.rs:80                 cx.sess().span_suggestion(spans.0,
src/returns.rs:81                                           "remove `return` as shown:",
src/returns.rs:82                                           snippet);
src/returns.rs:112:23: 113:76 error: no method named `span_note` found for type `&rustc::session::Session` in the current scope
src/returns.rs:112             cx.sess().span_note(note_span,
src/returns.rs:113                                 "this expression can be directly returned");
src/cyclomatic_complexity.rs:143:19: 145:119 error: no method named `span_note` found for type `&rustc::session::Session` in the current scope
src/cyclomatic_complexity.rs:143         cx.sess().span_note(span, &format!("Clippy encountered a bug calculating cyclomatic complexity \
src/cyclomatic_complexity.rs:144                                             (hide this message with `#[allow(cyclomatic_complexity)]`): \
src/cyclomatic_complexity.rs:145                                             cc = {}, arms = {}, div = {}. Please file a bug report.", cc, narms, div));
error: aborting due to 12 previous errors
Could not compile `clippy`.

To learn more, run the command again with --verbose.
```

```
coreyf@frewbook-pro ~/D/r/rust-clippy (master)> rustc -Vv
rustc 1.7.0-nightly (2370d461a 2015-12-30)
binary: rustc
commit-hash: 2370d461a648035b54aae492e7df66256f0f18c6
commit-date: 2015-12-30
host: x86_64-apple-darwin
release: 1.7.0-nightly
```

### PR DESCRIPTION
**Pull request by Manishearth**
Fixes #525

compiletest isn't working with the cyclomatic complexity lint, it's mentioning an unexpected error where the error marking exists. Not sure what's going on here, cc @oli-obk / @llogiq

I'm going to publish this anyway, since the code is working but we have a bug with the test harness.

### CODE REVIEW

File: tests/compile-fail/cyclomatic_complexity.rs [hunk 1]
```diff
@@ -4,7 +4,8 @@
 #![deny(cyclomatic_complexity)]
 #![allow(unused)]
 
-fn main() { //~ ERROR: The function has a cyclomatic complexity of 28.
+
+fn main() { //~ERROR The function has a cyclomatic complexity of 28
```
Comment (Manishearth) [inline tests/compile-fail/cyclomatic_complexity.rs]: (Updating the error marking to match the ones used in other tests doesn't help)

**Comment by oli-obk** [comment]
Not on my pc right now, but it might be related to the colon after ERROR or HINT, or then lack of a colon. Not sure what's causing it

**Comment by Manishearth** [comment]
Oh, I see the issue. The error is being emitted twice there.

**Comment by llogiq** [comment]
Why is it emitted twice?

**Pull request merged**